lib/query/compiler.js

```diff
@@ -574,6 +578,12 @@ assign(QueryCompiler.prototype, {
     );
   },
 
+  distinctOn(value) {
+    throw new Error(
+      '.distinctOn() is currently only supported on PostgreSQL'
```

Comment:
How difficult would it be to support equivalent MySQL syntax? https://stackoverflow.com/questions/17673457/converting-select-distinct-on-queries-from-postgresql-to-mysql

Also does that also work on redshift?

Reply:
@kibertoad I don't think it is a very good idea to emulate this feature in Knex because:

1. It introduces more complexity (both in implementation, and affects runtime performance) 
2. I don't see a straightforward emulation that is generically applicable 

For instance, consider the last example in that answer, if we add one more column (col5) to the order by then the solution will need more modification because we would need to find the minimum of col5 among all rows that have same value of col4 potentially introducing another level of nesting. 

Redshift doesn't support distinct on either.